inc/contract_item.class.php

```diff
@@ -632,8 +632,18 @@ static function showForContract(Contract $contract, $withtemplate = 0) {
                    || empty($data["name"])) {
                   $name = sprintf(__('%1$s (%2$s)'), $name, $objdata["id"]);
                }
-               $link = $itemtype::getFormURLWithID($objdata["id"]);
-               $name = "<a href=\"".$link."\">".$name."</a>";
+
+               $itemdata = new $item();
+               $itemdata->getFromDB($objdata['id']);
+               if (Session::haveRight($item::$rightname, $item::READMY)
```

Comment:
`$item` is not necessary a `Project`, so `$item::READMY` may not exists.

Why did you not simply use `if ($item->can($objdata['id'], READ)) { ... }` ?

Reply:
$item::READ may not exists too.
and for READMY right if must check if its an item affected to me.

Reply:
Unless I am wrong, `CommonDBTM::can($ID, READ)` will call `canView` and `canViewItem` methods, which should already contains this kind of checks. `READ` constant is a global constant, so it is always defined. `$item::READMY` does not exists for most of classes, so it will trigger a PHP error.

Reply:
You're right i always forget CommonDBTM::can.
PR changed